### PR TITLE
feat: Signup page

### PR DESCRIPTION
**Pull request by tusharkandpal**
- added Signup functionality
- added `useNotesOrLabelsPage` hook
- common `Auth.css` for Login & Signup components
- updated app icon

### CODE REVIEW

File: src/components/shared/sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -75,26 +78,33 @@ export function Sidebar() {
...
+        {isEitherNotesOrLabelsPage && (
+          <li
+            className="new-note-item"
+            onClick={() => {
+              displayDispatch({
+                type: "ADD_NOTE_TOGGLE",
+                payload: { addNoteToggle: true },
+              });
+              displayDispatch({ type: "SIDEBAR_TOGGLE" });
+            }}
```
Comment (JantuDeb) [inline src/components/shared/sidebar/Sidebar.jsx:90]: Create handler function for calling multiple function

**Review by JantuDeb — APPROVED** [review]
Overall the code looks fine

**Review by Kothalkarpriya — APPROVED** [review]
Good to merge

**Review by t007rushi — APPROVED** [review]
Look good to merge

**Pull request merged**